### DISCUSSION TITLE
Allow to export "empty" page middleware

### DISCUSSION THREAD

**Discussion by vklimontovich** [vercel/next.js#37346] [Ideas]
### Describe the feature you'd like to request

In some cases, next app will require middleware only if certain config param is enabled. For example, the app can record usage only if certain env variable is set.

To avoid execution of middleware that does nothing, it would be nice to allow export "null" or "undefined" middleware that signals Edge Framework to not run middleware at all.

### Describe the solution you'd like

If `_middleware.ts` export `undefined` or `null`, next should not run it at all
```typescript
import {NextMiddleware, NextResponse} from "next/server";

const middleware: NextMiddleware = (req, res) => {
    //do something
    return NextResponse.next()
}

export default process.env.FEATURE_THAT_REQUIRE_MIDDLEWARE_ENABLED ? middleware : null;
```

### Describe alternatives you've considered

Alternative solution would be like this:

```typescript
import {NextMiddleware, NextResponse} from "next/server";

const middleware: NextMiddleware = (req, res) => {
    if (process.env.FEATURE_THAT_REQUIRE_MIDDLEWARE_ENABLED) {
        //do something
    } else {
        return NextResponse.next()
    }
}

export default middleware;
```

However, in this case middleware is executed all the time, which could increase usage bills, response time etc

**Comment by matamatanot** [comment]
https://github.com/vercel/next.js/pull/36772

> declare your middleware in the root folder and use `NextRequest` parsed URL to define which path the middleware code should be executed for.

**Comment by vklimontovich** [comment]
@matamatanot. This is not going to help :( Middleware still will be executed, meaning Vercel charges etc
